File: adapters/webhook/node/example-agent.mjs
// A ten-line agent for Yui: every turn gets a screen back. Start it, then: node yui-webhook.mjs run --webhook http://127.0.0.1:8787
import { createServer } from "node:http";
createServer(async (req, res) => {
  let raw = "";
  for await (const chunk of req) raw += chunk;
  const turn = JSON.parse(raw);
  const taps = turn.messages.filter((m) => m.event?.echo).map((m) => m.event.echo); // taps on our screen
  const reply = taps.length ? `${taps.at(-1)} it is. Enjoy!` : 'Hi! What sounds good?\n```yui\nchoose "Pick one" Coffee|Walk|Nap\n```';
  res.writeHead(200, { "content-type": "application/json" }).end(JSON.stringify({ reply }));
}).listen(process.env.PORT ?? 8787, "127.0.0.1");

File: adapters/webhook/python/example_agent.py
# A ten-line agent for Yui: every turn gets a screen back. Start it, then: yui_webhook.py run --webhook http://127.0.0.1:8787
import json
from http.server import BaseHTTPRequestHandler, HTTPServer
class Agent(BaseHTTPRequestHandler):
    def do_POST(self):
        turn = json.loads(self.rfile.read(int(self.headers["content-length"])))
        taps = [m["event"]["echo"] for m in turn["messages"] if (m["event"] or {}).get("echo")]  # taps on our screen
        reply = f"{taps[-1]} it is. Enjoy!" if taps else 'Hi! What sounds good?\n```yui\nchoose "Pick one" Coffee|Walk|Nap\n```'
        self.send_response(200); self.send_header("content-type", "application/json"); self.end_headers()
        self.wfile.write(json.dumps({"reply": reply}).encode())
HTTPServer(("127.0.0.1", int(__import__("os").environ.get("PORT", 8787))), Agent).serve_forever()
